kratos/utilities/python_function_callback_utility.h

```diff
@@ -8,214 +8,357 @@
+        const double X = 0.0,
+        const double Y = 0.0,
+        const double Z = 0.0
+        )
     {
         array_1d<double,3> xglobal;
         xglobal[0] = x;
         xglobal[1] = y;
         xglobal[2] = z;
-        array_1d<double,3> xlocal = prod(mR, (xglobal - mxc) );
-        return CallFunction(xlocal[0],xlocal[1],xlocal[2],t);
+        array_1d<double,3> xlocal = prod(mRotationMatrix, (xglobal - mCenterCoordinates) );
+        array_1d<double,3> xglobal_initial;
+        xglobal_initial[0] = X;
+        xglobal_initial[1] = Y;
+        xglobal_initial[2] = Z;
+        array_1d<double,3> xlocal_initial = prod(mRotationMatrix, (xglobal_initial - mCenterCoordinates) );
+        return CallFunction(xlocal[0],xlocal[1],xlocal[2], t,xlocal_initial[0],xlocal_initial[1],xlocal_initial[2]);
     }
 
-    double CallFunction(const double x, const double y, const double z, const double t)
+    /**
+     * @brief This calls the evaluation function
+     * @param x The x coordinate
+     * @param y The y coordinate
+     * @param z The z coordinate
+     * @param t The time variable
+     * @param X The initial x coordinate
+     * @param Y The initial y coordinate
+     * @param Z The initial z coordinate
+     */
+    double CallFunction(
+        const double x,
+        const double y,
+        const double z,
+        const double t,
+        const double X = 0.0,
+        const double Y = 0.0,
+        const double Z = 0.0
+        )
     {
-        main_namespace["x"] = x;
-        main_namespace["y"] = y;
-        main_namespace["z"] = z;
-        main_namespace["t"] = t;
-
-
-//         #if PY_MAJOR_VERSION >= 3
-//         PyObject* res = PyEval_EvalCode(mbytecode.ptr(),main_namespace.ptr(),main_namespace.ptr());
-//         #else
-//         PyObject* res = PyEval_EvalCode((PyCodeObject*)(mbytecode.ptr()),main_namespace.ptr(),main_namespace.ptr());
-//         #endif
-        return pybind11::eval(mfunction_body, main_namespace).cast<double>();
+        mMainNameSpace["x"] = x;
+        mMainNameSpace["y"] = y;
+        mMainNameSpace["z"] = z;
+        mMainNameSpace["X"] = X;
+        mMainNameSpace["Y"] = Y;
+        mMainNameSpace["Z"] = Z;
+        mMainNameSpace["t"] = t;
+
+        return pybind11::eval(mFunctionBody, mMainNameSpace).cast<double>();
     }
 
+    ///@}
 
 private:
-    pybind11::object main_module;
-    pybind11::object main_namespace;
-    std::string mfunction_body;
-    pybind11::object mbytecode;
 
-    bool mdepends_on_space = true;
-    bool muse_local_system = false;
-    BoundedMatrix<double, 3, 3> mR;
-    array_1d<double, 3> mxc;
-};
+    ///@name Member Variables
+    ///@{
+
+    pybind11::object mMainModule;       /// The main python module
+    pybind11::object mMainNameSpace;    /// The main python namespace (the variables considered on the python function)
+    const std::string mFunctionBody;    /// The function body
+    pybind11::object mByteCode;         /// Some byte code
```

Comment:
This is not used, I think it can be removed

Reply:
True